File: backend/src/Models/Page.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

public class Page
{
    [Key]
    public Guid Id { get;  set; }
    
    public string Title { get;  set; }
    
    public string Description { get;  set; }
    
    public string Picture { get;  set; }
    
    [ForeignKey("UserId")]
    public User User { get; set; }
    
    public List<Link> Links { get; set; } = new List<Link>();
    
    public Page(){}
    public Page(string title, string description, string picture, User user)
    {
        Title = title;
        Description = description;
        Picture = picture;
        User = user;
    }
}

File: mytree/src/main/java/com/mytree/mytree/model/Page.java
package com.mytree.mytree.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Page {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String title;

    private String description;

    @ManyToOne()
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(cascade = CascadeType.ALL)
    private List<Link> links;

    public Page(String title, String description, User user) {
        this.title = title;
        this.description = description;
        this.user = user;
    }
}
